src/System.Linq/src/System/Linq/Aggregate.cs

```diff
@@ -11,33 +10,77 @@ public static partial class Enumerable
     {
         public static TSource Aggregate<TSource>(this IEnumerable<TSource> source, Func<TSource, TSource, TSource> func)
         {
-            if (source == null) throw Error.ArgumentNull("source");
-            if (func == null) throw Error.ArgumentNull("func");
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            if (func == null)
+            {
+                throw Error.ArgumentNull(nameof(func));
+            }
+
             using (IEnumerator<TSource> e = source.GetEnumerator())
             {
-                if (!e.MoveNext()) throw Error.NoElements();
```

Comment:
Very minor, very subjective, but perhaps worth mentioning here more than any other changes to this project; `e` does not strike me as a descriptive name, and all the more so since it's often used for exceptions. Even `en` would be an improvement IMO.